src/vmEntry.cpp

```diff
@@ -419,6 +419,7 @@ void JNICALL VM::VMInit(jvmtiEnv* jvmti, JNIEnv* jni, jthread thread) {
 }
 
 void JNICALL VM::VMDeath(jvmtiEnv* jvmti, JNIEnv* jni) {
+    Profiler::instance()->set_jvm_dying();
```

Comment:
Why in `Profiler`? You're in the `VM` class, and the flags makes sense for the VM.